### PR TITLE
Send updates for all csv rows, even if there were errors for previous rows

### PR DESCRIPTION
**Pull request by rkondratowicz**

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/slc/services/UpdateLetterStatusJob.java [hunk 1]
```diff
@@ -46,8 +49,15 @@ public void run() {
                 .map(this::tryParse)
                 .filter(Objects::nonNull)
                 .forEach(parsedReport -> {
-                    parsedReport.statuses.forEach(sendLetterClient::updatePrintedAt);
-                    ftpClient.deleteReport(parsedReport.path);
+                    List<Boolean> successStatuses = parsedReport
+                        .statuses
+                        .stream()
+                        .map(this::trySendUpdate)
+                        .collect(toList());
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/slc/services/UpdateLetterStatusJob.java:56]: would it be better to just continue with the stream here avoiding collection and essentially evaluation of the list?

**Review by doncem — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/slc/services/UpdateLetterStatusJob.java (hunk 1 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/slc/services/UpdateLetterStatusJob.java:56]: I can't, `allMatch` will stop going over the stream on first `false`

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/slc/services/UpdateLetterStatusJob.java:56]: 😞. how about using `.reduce` function instead of `.map`?

**Comment by doncem** [comment]
I would prefer to avoid evaluating as a list and streaming over it again. Nothing wrong conceptually so unblocking

**Review by doncem — APPROVED** [review]

**Pull request merged**